### PR TITLE
Added whitelist and blacklist capabilities to HiveDatasetFinder.

### PR DESCRIPTION
**Pull request by ibuenros**

### CODE REVIEW

**Comment by ibuenros** [comment]
@ydai1124 can you review? Thanks!

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/WhitelistBlacklist.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  private boolean accept(String db, Optional<String> table) {
+    if (!this.blacklistMultimap.isEmpty() && multimapContains(this.blacklistMultimap, db, table, true)) {
+      return false;
+    }
+
+    return this.whitelistMultimap.isEmpty() || multimapContains(this.whitelistMultimap, db, table, false);
+  }
+
+  private void populateMultimap(SetMultimap<Pattern, Pattern> multimap, String list) throws IOException {
+    Splitter tokenSplitter = Splitter.on(",");
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/WhitelistBlacklist.java]: You may want to `trimResults` and `omitEmptyStrings` for these splitters.

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java [hunk 2]
```diff
@@ -49,37 +54,63 @@
...
-    this.tablePattern = properties.getProperty(TABLE_PATTERN_KEY, DEFAULT_TABLE_PATTERN);
-    this.properties = properties;
+    String whitelistKey = HIVE_DATASET_PREFIX + "." + WhitelistBlacklist.WHITELIST;
+    Preconditions.checkArgument(properties.containsKey(DB_KEY) || properties.containsKey(whitelistKey),
+        String.format("Must specify %s or %s.", DB_KEY, whitelistKey));
+
+    Config config = ConfigFactory.parseProperties(properties);
+
+    if (properties.containsKey(DB_KEY)) {
+      this.whitelistBlacklist = new WhitelistBlacklist(this.properties.getProperty(DB_KEY) + "."
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java]: Why not making blacklist a specific conf key?

Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java]: It is. This block is for backwards compatibility. However, the intended use is to set `hive.dataset.whitelist` and `hive.dataset.blacklist` which is taken care of by the else block. (See unit test)

Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java]: I see.

**Comment by ydai1124** [comment]
LGTM. One minor comment on the Splitters.

**Pull request merged**